vortex-ipc/src/reader.rs

```diff
@@ -202,19 +202,10 @@ impl<'a, R: Read> StreamArrayReader<'a, R> {
                 continue;
             }
 
-            // TODO(@jdcasale): replace this with compute scalar_sum when we've added it
             let indices_for_batch = slice(indices, left, right)?.flatten_primitive()?;
-            let shifted = match_each_integer_ptype!(indices_for_batch.ptype(), |$P| {
-                let shifted = indices_for_batch
-                    .typed_data::<$P>()
-                    .iter()
-                    .map(|&idx| {
-                        idx as u64 - row_offset as u64
-                    })
-                    .collect_vec();
-                PrimitiveArray::from(shifted)
+            let shifted_arr = match_each_integer_ptype!(indices_for_batch.ptype(), |$T| {
+                indices_for_batch.scalar_subtract(&Scalar::from(row_offset as $T))?
```

Comment:
You should use the entry point function you defined. `scalar_subtract::(indices_for_batch, row_offset)`. The whole point of the `T: Into<Scalar>` is so you can pass your usize and it will be casted into whatever type indices_for_batch is.